src/components/Header/Header.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  todoTitle: string;
+  setTodoTitle: (title: string) => void;
+  formSubmit: (event: React.FormEvent) => void;
+  isAdding: boolean;
+  inputRef: React.RefObject<HTMLInputElement>;
+  updateTodo: (
+    todoId: number,
+    title: string,
+    completed?: boolean,
+  ) => Promise<void> | undefined;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  todoTitle,
+  setTodoTitle,
+  formSubmit,
+  isAdding,
+  inputRef,
+  updateTodo,
+}) => {
+  const allTodosCompleted = todos.every(todo => todo.completed);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleCompleteAll = async () => {
+    const allComplete = todos.every(todo => todo.completed);
```

Comment:
`areAllTodosCompleted` name will describe it better
But you can use value from the scope